### PR TITLE
adding autogenerated_domain_namelabel_scope field

### PR DESCRIPTION
**Pull request by angouloure**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@zhoxing-ms please help with review.

File: src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py [hunk 1]
```diff
@@ -243,6 +243,41 @@ def test_container_create_with_msi_scope(self, resource_group, resource_group_lo
                          self.exists('ipAddress.ip'),
                          self.check('containers[0].image', '{image}')])
 
+
+    # Test create container using auto generated domain name label scope.
+    @live_only()
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py:248]: Why does this test not support recording mode?

File: src/azure-cli/azure/cli/command_modules/container/_params.py [hunk 2]
```diff
@@ -66,6 +66,7 @@ def load_arguments(self, _):
         c.argument('ports', type=int, nargs='+', default=[80], help='A list of ports to open. Space-separated list of ports')
         c.argument('protocol', arg_type=get_enum_type(ContainerNetworkProtocol), help='The network protocol to use')
         c.argument('dns_name_label', help='The dns name label for container group with public IP')
+        c.argument('auto_generated_domain_name_label_scope', arg_type=get_enum_type(AutoGeneratedDomainNameLabelScope), help='The autogenerate name label policy chosen by user')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/container/_params.py:69]: The name of this parameter seems too long for users to input. Could you give it a short alias ?

**Comment by zhoxing-ms** [comment]
> ImportError: cannot import name 'AutoGeneratedDomainNameLabelScope' from 'azure.mgmt.containerinstance.models' 

Please note the model `AutoGeneratedDomainNameLabelScope` is not included in the SDK package version `azure-mgmt-containerinstance==9.1.0`. Please bump the SDK version to https://pypi.org/project/azure-mgmt-containerinstance/9.2.0/

File: src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py (hunk 1 above)
Comment (angouloure) [reply src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py:248]: How do we make it support recording mode?

Comment (joseph-porter) [reply src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py:248]: @angouloure You'll just have to remove `@live_only()`. Record is the default behavior.

The test documentation is here: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md

I can walk you through our tests if you have any issues.

**Comment by wangzelin007** [comment]
@angouloure,
Please re-record all these failed tests and commit the new recording files.
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_create src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_create_with_auto_gen_dnl_scope
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_create_with_acr
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_create_with_msi
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_create_with_zone
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_git_repo_volume_mount
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_create_with_vnet
src/azure-cli/azure/cli/command_modules/container/tests/latest/test_container_commands.py::AzureContainerInstanceScenarioTest::test_container_export

**Pull request closed without merging**